core/src/main/java/com/linecorp/armeria/client/pool/DefaultKeyedChannelPool.java

```diff
@@ -280,6 +287,12 @@ protected boolean offerChannel(K key, Channel channel) {
 
     @Override
     public void close() {
+        for (Channel ch : createdChannels) {
+            if (ch.isOpen()) {
+                ch.close().syncUninterruptibly();
+            }
+        }
+        createdChannels.clear();
         for (Iterator<Deque<Channel>> i = pool.values().iterator(); i.hasNext();) {
```

Comment:
It seems like we do not need to iterate over `pool` anymore because we have all the channels in `pool` also in `createdChannels`.

Instead, we need to iterate over `pendingConnections` and close them as soon as they are connected.

This introduces the risk of `close() being taking too long depending on network condition, so we may want to allow closing the channels in `pendingConnections` asynchronously as an exception.